library/Network/CloudSeeder/Main.hs

```diff
@@ -114,11 +124,16 @@ cli (DeployStack nameToDeploy) config = do
 
   templateBody <- readFile $ nameToDeploy <> ".yaml"
 
+  let decodeOrFailure = decodeEither (encodeUtf8 templateBody) :: Either String Value
+  template <- either (throwing _CliTemplateDecodeFail) return decodeOrFailure
+  let requiredParams = template ^. key "Parameters" . _Object.to keys
+  when ("Env" `notElem` requiredParams) (throwing _CliTemplateMissingEnvParameter ())
+
   maybeOutputs <- mapM (\stackName -> (stackName,) <$> getStackOutputs (mkStackName stackName)) dependencies
   let outputsOrFailure = runErrors $ traverse (extractResult (flip const)) maybeOutputs
-  outputs <- either (throwing _CliMissingDependencyStacks) return outputsOrFailure
+  outputs <- either (throwing _CliMissingDependencyStacks) (return . concat) outputsOrFailure
 
-  let parameters = envVars ++ concat outputs
+  let parameters = filter (\(k,_) -> k `elem` requiredParams) (outputs ++ envVars)
```

Comment:
I might break this lambda out into a separate function to make it a little more clear what’s going on here. Something like this:

```haskell
let isRequired = (`elem` requiredParams)
    parameters = filter (isRequired . fst) (outputs ++ envVars)
```